server/form-pages/apply/offence-and-licence-information/hdc-licence-and-cpp-details/hdcLicenceDates.ts

```diff
@@ -0,0 +1,62 @@
+import { TaskListErrors } from '@approved-premises/ui'
+import { Cas2Application as Application } from '@approved-premises/api'
+import type { ObjectWithDateParts } from '@approved-premises/ui'
+import { Page } from '../../../utils/decorators'
+import TaskListPage from '../../../taskListPage'
+import { nameOrPlaceholderCopy } from '../../../../utils/utils'
+import { getQuestions } from '../../../utils/questions'
+import { dateAndTimeInputsAreValidDates, DateFormats } from '../../../../utils/dateUtils'
+import { dateBodyProperties } from '../../../utils'
+
+type HDCLicenceDatesBody = ObjectWithDateParts<'hdcEligibilityDate'> & ObjectWithDateParts<'conditionalReleaseDate'>
+
+@Page({
+  name: 'hdc-licence-dates',
+  bodyProperties: [...dateBodyProperties('hdcEligibilityDate'), ...dateBodyProperties('conditionalReleaseDate')],
+})
+export default class HDCLicenceDates implements TaskListPage {
+  documentTitle = `The person's Home Detention Curfew (HDC) licence dates`
+
+  personName = nameOrPlaceholderCopy(this.application.person)
+
+  title = `${this.personName}'s Home Detention Curfew (HDC) licence dates`
+
+  questions = getQuestions(this.personName)['hdc-licence-and-cpp-details']['hdc-licence-dates']
+
+  options: Record<string, string>
+
+  body: HDCLicenceDatesBody
+
+  constructor(
+    body: Partial<HDCLicenceDatesBody>,
+    private readonly application: Application,
+  ) {
+    this.body = body as HDCLicenceDatesBody
+  }
+
+  previous() {
+    return 'taskList'
+  }
+
+  next() {
+    return 'cpp-details'
+  }
+
+  response() {
+    return {
+      'HDC eligibility date': DateFormats.dateAndTimeInputsToUiDate(this.body, 'hdcEligibilityDate'),
+      'Conditional release date': DateFormats.dateAndTimeInputsToUiDate(this.body, 'conditionalReleaseDate'),
```

Comment:
Have we been asked to make any other validation tests on these dates? e.g.

- do they both need to be in the future?
- does the release date need to be later that the HDC eligibility date?

I'm not suggesting that this is necessary or ought to be done now, but maybe there needs to be a conversation which might result in a "ticket" for further work?